src/plugins/event-plugins/NavigationPlugin.ts

```diff
@@ -1,20 +1,35 @@
 import { InternalPlugin } from '../InternalPlugin';
 import { NavigationEvent } from '../../events/navigation-event';
 import { PERFORMANCE_NAVIGATION_EVENT_TYPE } from '../utils/constant';
+import { defaultPerformanceIgnore } from '../utils/performance-utils';
 
 export const NAVIGATION_EVENT_PLUGIN_ID = 'navigation';
 
 const NAVIGATION = 'navigation';
 const LOAD = 'load';
 
+export type PartialNavigationPluginConfig = {
+    ignore?: (event: PerformanceEntry) => any;
+};
+
+export type NavigationPluginConfig = {
+    ignore: (event: PerformanceEntry) => any;
+};
+
+export const defaultNavigationPluginConfig = {
+    ignore: defaultPerformanceIgnore
+};
```

Comment:
suggestion: Rename these types like `PerformancePluginConfig` and move them to `performance-utils` where they can be shared.

This should be non-breaking, because we don't export this type.

Reply:
I moved all of the shared performance stuff to PerformanceConfig in performance-utils. I should not that `eventLimit` should be `resourceEventLimit`, but changing that at this point would be breaking I think.